src/libs/ui/src/molecules/PlpAccordians/index.tsx

```diff
@@ -0,0 +1,230 @@
+import LabelAccordion from '../PlpLabelAccordian';
+import CheckboxAccordion from '../PlpCheckBoxAccordian';
+import RadioAccordion from '../PlpRadioAccordian';
+import Modal from '../PlpSortModal';
+
+interface FilterData {
+  productCategory: string[];
+  bestSeller: string[];
+  productTypes: string[];
+  skinConcern: string[];
+  skinType: string[];
+}
+
+interface SidebarProps {
+  filterData: FilterData;
+  className?: string; // Classname for outer div
+  ulClassName?: string; // Classname for <ul>
+  liClassName?: string; // Classname for <li>
+  onBestSellerChange: (checked: boolean) => void; // New prop for best seller filter
+  onCategorySelect: (category: string | null) => void; // Update type to allow null
+  onSortChange: (sortOption: string) => void;
+  enableBestSeller: boolean; // Prop for sort change
+}
+
+const PlpAccordians: React.FC<SidebarProps> = ({
+  filterData,
+  className,
+  ulClassName,
+  liClassName,
+  onBestSellerChange,
+  onCategorySelect,
+  onSortChange,
+  enableBestSeller
+}) => {
+  
+
+  const [showAllCategories, setShowAllCategories] = useState(false);
+  const [selectedProductCategory, setSelectedProductCategory] = useState<string | null>(null);
+  const [checkedFilters, setCheckedFilters] = useState<{ [key: string]: boolean }>({});
+  const [isSmallScreen, setIsSmallScreen] = useState<boolean>(false);
+  const [selectedSort, setSelectedSort] = useState<string>(''); // State for sort option
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false); // State for modal visibility
+
+  // Responsive check for small screens
+  useEffect(() => {
+    const handleResize = () => {
+      setIsSmallScreen(window.innerWidth <= 1020);
+    };
+
+    window.addEventListener('resize', handleResize);
+    handleResize(); // Initial check
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+
+  }, []);
+
+  // Toggle category visibility
+  const toggleShowCategories = () => {
+    setShowAllCategories(!showAllCategories);
+  };
+
+  // Handle product category click
+  const handleProductCategoryClick = (category: string) => {
+    console.log('category', category);
+    const isSelected = category === selectedProductCategory;
+    setSelectedProductCategory(isSelected ? null : category);
+    onCategorySelect(isSelected ? null : category);
+  };
+
+  // Handle checkbox changes
+  const handleCheckboxChange = (option: string) => {
+    setCheckedFilters((prev) => ({
+      ...prev,
+      [option]: !prev[option],
+    }));
+
+    if (option === 'Best Seller') {
+      onBestSellerChange(!checkedFilters[option]);
+    }
+  };
+
+  // Handle sort option change
+  const handleSortChange = (sortOption: string) => {
+    setSelectedSort(sortOption);
+    onSortChange(sortOption);
+  };
+
+  // Sort options
+  const sortOptions = ['Price: Low to High', 'Price: High to Low', 'New Arrivals'];
+
+  // Define checkbox accordions
+  const checkboxAccordions = [
```

Comment:
Please remove this checkboxAccordions variable from here, add to top of the file

Reply:
fixed